stdlib/universe/table_fns.go

```diff
@@ -98,6 +97,13 @@ func tableFind(ctx context.Context, to *flux.TableObject, fn *execute.TablePredi
 	if p, ok := p.(lang.LoggingProgram); ok {
 		p.SetLogger(deps.Logger)
 	}
+
+	// If there is an operator profiler in the execution options, and it is not
+	// in the context standalone, add it.
+	if deps.ExecutionOptions.OperatorProfiler != nil {
```

Comment:
You're right, it will get injected more than once for recursive tableFind calls. We could put a guard on it. Although, I think using the context directly for accessing the profiler is leading to this awkwardness. I'm thinking we could take the profiler from the newly created `ExecutionOptions` struct. The dependencies store a *pointer* to this and we can updated it in the interpreter. It immediately becomes available during execution then. No need for extra injections.

Reply:
The main issue with the context is that it is only useful for passing values down the call chain. It can't really function as a state that lives for the life of the query. It can pass values from interpreter phase to the top-level execution phase, but it can't pass values between interpreted functions. Since we now have interpreted functions that trigger execution, we can't rely on context to pass values from interpreter to deeper executions such as `tableFind`.